### PR TITLE
Implicitly require https scheme for ca-url cmd line arg

### LINKED ISSUE #336: Unexpected behaviour when missing https from CA URL
### Subject of the issue

Unexpected behaviour and unclear resolution when generating a token for mTLS cert creation. This was caused by a missing `https://` from the CA URL.

In the example, I'm using an offline private key to auto-provision a token, from which I can retrieve a certificate.

### Your environment
* OS - macOS (client) / GCP COS (CA server)
* Version - 0.14.6 and 51b9867

### Steps to reproduce

To start, I ran `rm -rf ~/.step`, to ensure this is repeatable.

#### Success case:

```
matt@Matts-MBP:~/ha > TOKEN=$(step ca token ha-lambda-1 --provisioner LambdaAWS --key lambda-jwk.unencrypted --ca-url https://ca.example.com --root ./root_ca.crt)
✔ Provisioner: LambdaAWS (JWK) [kid: 8whbKJXYCv3QnhhttAUJTrCK1-oESe7jpaxxxxxT2Hw]
matt@Matts-MBP:~/ha > echo $TOKEN
eyJhb ... qcJg
matt@Matts-MBP:~/ha > step ca certificate ha-lambda-1 /tmp/step.crt /tmp/step.key --token "$TOKEN" --ca-url ca.example.com
✔ CA: ca.example.com
✔ Certificate: /tmp/step.crt
✔ Private Key: /tmp/step.key
```

#### Fail case:

The only difference is missing `https://` from the `--ca-url` parameter.

```
matt@Matts-MBP:~/ha > TOKEN=$(step ca token ha-lambda-1 --provisioner LambdaAWS --key lambda-jwk.unencrypted --ca-url ca.example.com --root ./root_ca.crt)
✔ Provisioner: LambdaAWS (JWK) [kid: 8whbKJXYCv3QnhhttAUJTrCK1-oESe7jpaxxxxxT2Hw]
matt@Matts-MBP:~/ha > echo $TOKEN
eyJhb ... xQog
matt@Matts-MBP:~/ha > step ca certificate ha-lambda-1 /tmp/step.crt /tmp/step.key --token "$TOKEN" --ca-url ca.example.com
The request lacked necessary authorization to be completed. Please see the certificate authority logs for more info.
Re-run with STEPDEBUG=1 for more info.
```

The CA logs are unclear on the problem and resolution, I did a lot of reading code and hacking around to figure this one out 😄 

```
time="2020-07-19T22:03:45Z" level=warning duration="598.741µs" duration-ns=598741 error="authority.Authorize: authority.authorizeSign: authority.authorizeToken: provisioner not found or invalid audience (https:///1.0/sign)" fields.time="2020-07-19T22:03:45Z" method=POST name=ca ott=eyJh ... xQog path=/sign protocol=HTTP/2.0 referer= remote-address=212.x.x.x request-id=bsac6gf8uu4dps7v7ekg size=144 status=401 user-agent="Smallstep CLI/0.14.6 (darwin/amd64)" user-id=
```

### PR DESCRIPTION
**Pull request by dopey**
Fixes #336

### CODE REVIEW

File: utils/cautils/token_flow.go [hunk 1]
```diff
@@ -31,11 +31,36 @@ const (
 	SSHRekeyType
 )
 
-// parseAudience creates the ca audience url from the ca-url
-func parseAudience(ctx *cli.Context, tokType int) (string, error) {
+// CtxCAURL gets the ca-url from the command context. Prepend an 'https' scheme
+// if the URL does not have a scheme. Error if the URL scheme is not implicitly
+// 'https'.
+func CtxCAURL(ctx *cli.Context, isRequired bool) (string, error) {
```
Comment (maraino) [inline utils/cautils/token_flow.go:37]: I feel this method makes more sense in the flags package, `flags.ParseCaURL`?

File: command/ca/revoke.go [hunk 2]
```diff
@@ -306,8 +306,11 @@ func (f *revokeFlow) getClient(ctx *cli.Context, serial, token string) (cautils.
 	}
 
 	// Create online client
+	caURL, err := cautils.CtxCAURL(ctx, false)
```
Comment (maraino) [inline command/ca/revoke.go:309]: It looks the ca-url is required here too, it's used below in the client.

Comment (maraino) [reply command/ca/revoke.go:309]: If this is the case is probably required always.

**Comment by dopey** [comment]
The ssh commands all use the same token_flow methods as the x509 commands.

**Comment by dopey** [comment]
I've written unit tests and done some prelim command line tests for `step ca token`, `step ca certificate`, and `step ssh certificate`. All seem to work as expected.

**Pull request merged**